src/pjpersist/serialize.py

```diff
@@ -547,6 +549,7 @@ def get_non_persistent_object(self, state, obj):
 
         # this method must NOT change the passed in state dict
         state = dict(state)
+        setState = True
```

Comment:
Why do you need this variable?

Reply:
because L571 will figure whether it needs to set state, the other cases always want to set state

either this or I could add a private method to set state...

Reply:
Isn't the purpose of this MR is to always set state? When we don't want to do that?

Reply:
We don't want to set state when `state` has `_py_factory` and `_py_factory_args` but nothing else. That would 'clean out' the object, resetting state that was set by `_py_factory_args`

Reply:
I don't get it.  Why would it "clean out" the object? What does it have to do with `_py_factory_args`? If empty `state` somehow overrides `_py_factory_args`, wouldn't non-empty state also override it? Why `_py_factory_args` is there at all, if all state can be stored in `state` itself?

Reply:
Example:
```
state={
    '_py_factory': 'PersistentDict',
    '_py_factory_args': {some_dict_items}
}

            factory = self.simple_resolve(state.pop('_py_factory'))
            factory_args = self.get_object(state.pop('_py_factory_args'), obj)
            sub_obj = factory(*factory_args)
```
now `sub_obj` has `some_dict_items`, `state` is `{}`

if we let the code run into
```
sub_obj_state = self.get_object(state, obj)
sub_obj.__setstate__(sub_obj_state)
```

`sub_obj` will get `some_dict_items` wiped by the 'empty' state

pretty obscure edge case, `doctest_ObjectReader_get_non_persistent_object_py_factory` failed on that

Reply:
Wait, so `sub_obj` will wipe itself? Isn't it the problem of `sub_obj`, and not the pjpersist?